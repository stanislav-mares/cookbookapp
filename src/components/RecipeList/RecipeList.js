import React from 'react';
import PropTypes from 'prop-types';

//components
import Recipe from 'components/Recipe/Recipe';

const RecipeList = (props) => {

  //mapp recipe list based on user input
  let filteredRecipes = [];

  (props.recipes ? props.recipes: []).forEach((recipe, i) => {
    if (recipe.title.toLowerCase().search(props.filterText.toLowerCase()) === -1) {
      return;
    }

    filteredRecipes.push(
      <li key={i}>
        <Recipe
          {...recipe}
          handleEditRecipe={props.handleEditRecipe}
          handleRemoveRecipe={props.handleRemoveRecipe}
        />
      </li>
    );
  });

  return(
    <div>
      <ul>
        {filteredRecipes.length > 0 ? filteredRecipes : 'No recipes'}
      </ul>
    </div>
  );
}

RecipeList.propTypes = {
  recipes: PropTypes.arrayOf(PropTypes.shape({
     id: PropTypes.number,
     title : PropTypes.string,
     desc : PropTypes.string,
     ingredients : PropTypes.arrayOf(PropTypes.string),
     procedure: PropTypes.arrayOf(PropTypes.string),
     avatarURL: PropTypes.string,
     likes : PropTypes.number,
     dislikes : PropTypes.number
  })),
  filterText: PropTypes.string,
  handleEditRecipe: PropTypes.func,
  handleRemoveRecipe: PropTypes.func
};

export default RecipeList;
