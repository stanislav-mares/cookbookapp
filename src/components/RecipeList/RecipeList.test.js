import React from 'react';
import enzymeConfig from 'testConfig/enzymeConfig';
import { shallow, mount } from 'enzyme';


import RecipeList from 'components/RecipeList/RecipeList';

describe('<RecipeList />', () => {
  it('Check component if there is no recipe', () => {
    const component = shallow(
      <RecipeList
        handleEditRecipe={() => {}}
        handleRemoveRecipe={() => {}}
      />
    );

    const items = component.find('li');
    expect(items.length).toBe(0);
    expect(component).toMatchSnapshot();
  });

  it('Check if rendering recipes works correctly', () => {
    const recipes = [
      {
        id: 1,
        title : "Spaghetti",
        desc : "Preparation time: 30min, Difficulty: 3/5",
        ingredients : ["one", "two", "three", "four", "five"],
        procedure: ["step_one", "step_two", "step_three", "step_four", "step_five"],
        avatarURL: '',
        likes : 4,
        dislikes : 4
      },
      {
        id: 2,
        title : "Tomato soup",
        desc : "Preparation time: 20min, Difficulty: 2/5",
        ingredients : ["one", "two", "three", "four", "five"],
        procedure: ["step_one", "step_two", "step_three", "step_four", "step_five"],
        avatarURL: '',
        likes : 5,
        dislikes : 5
      }
    ];

    const component = shallow(
      <RecipeList
        filterText=""
        recipes={recipes}
        handleEditRecipe={() => {}}
        handleRemoveRecipe={() => {}}
      />
    );

    const items = component.find('li');
    expect(items.length).toBe(recipes.length);
    expect(component).toMatchSnapshot();
  });

  it('Check if filtering list of recipes works correctly', () => {
    const recipes = [
      {
        id: 1,
        title : "Spaghetti",
        desc : "Preparation time: 30min, Difficulty: 3/5",
        ingredients : ["one", "two", "three", "four", "five"],
        procedure: ["step_one", "step_two", "step_three", "step_four", "step_five"],
        avatarURL: '',
        likes : 4,
        dislikes : 4
      },
      {
        id: 2,
        title : "Tomato soup",
        desc : "Preparation time: 20min, Difficulty: 2/5",
        ingredients : ["one", "two", "three", "four", "five"],
        procedure: ["step_one", "step_two", "step_three", "step_four", "step_five"],
        avatarURL: '',
        likes : 5,
        dislikes : 5
      }
    ];

    const component = shallow(
      <RecipeList
        filterText="Tomato soup"
        recipes={recipes}
        handleEditRecipe={() => {}}
        handleRemoveRecipe={() => {}}
      />
    );

    const items = component.find('li');
    expect(items.length).toBe(1);
    expect(component).toMatchSnapshot();
  });
});
