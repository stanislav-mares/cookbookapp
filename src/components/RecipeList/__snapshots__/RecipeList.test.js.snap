// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> Check component if there is no recipe 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeList
    handleEditRecipe={[Function]}
    handleRemoveRecipe={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        No recipes
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "No recipes",
      },
      "ref": null,
      "rendered": "No recipes",
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          No recipes
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "No recipes",
        },
        "ref": null,
        "rendered": "No recipes",
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<RecipeList /> Check if filtering list of recipes works correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeList
    filterText="Tomato soup"
    handleEditRecipe={[Function]}
    handleRemoveRecipe={[Function]}
    recipes={
      Array [
        Object {
          "avatarURL": "",
          "desc": "Preparation time: 30min, Difficulty: 3/5",
          "dislikes": 4,
          "id": 1,
          "ingredients": Array [
            "one",
            "two",
            "three",
            "four",
            "five",
          ],
          "likes": 4,
          "procedure": Array [
            "step_one",
            "step_two",
            "step_three",
            "step_four",
            "step_five",
          ],
          "title": "Spaghetti",
        },
        Object {
          "avatarURL": "",
          "desc": "Preparation time: 20min, Difficulty: 2/5",
          "dislikes": 5,
          "id": 2,
          "ingredients": Array [
            "one",
            "two",
            "three",
            "four",
            "five",
          ],
          "likes": 5,
          "procedure": Array [
            "step_one",
            "step_two",
            "step_three",
            "step_four",
            "step_five",
          ],
          "title": "Tomato soup",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
          <Recipe
            avatarURL=""
            desc="Preparation time: 20min, Difficulty: 2/5"
            dislikes={5}
            handleEditRecipe={[Function]}
            handleRemoveRecipe={[Function]}
            id={2}
            ingredients={
              Array [
                "one",
                "two",
                "three",
                "four",
                "five",
              ]
            }
            likes={5}
            procedure={
              Array [
                "step_one",
                "step_two",
                "step_three",
                "step_four",
                "step_five",
              ]
            }
            title="Tomato soup"
          />
        </li>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <Recipe
              avatarURL=""
              desc="Preparation time: 20min, Difficulty: 2/5"
              dislikes={5}
              handleEditRecipe={[Function]}
              handleRemoveRecipe={[Function]}
              id={2}
              ingredients={
                Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ]
              }
              likes={5}
              procedure={
                Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ]
              }
              title="Tomato soup"
            />
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Recipe
              avatarURL=""
              desc="Preparation time: 20min, Difficulty: 2/5"
              dislikes={5}
              handleEditRecipe={[Function]}
              handleRemoveRecipe={[Function]}
              id={2}
              ingredients={
                Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ]
              }
              likes={5}
              procedure={
                Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ]
              }
              title="Tomato soup"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "avatarURL": "",
              "desc": "Preparation time: 20min, Difficulty: 2/5",
              "dislikes": 5,
              "handleEditRecipe": [Function],
              "handleRemoveRecipe": [Function],
              "id": 2,
              "ingredients": Array [
                "one",
                "two",
                "three",
                "four",
                "five",
              ],
              "likes": 5,
              "procedure": Array [
                "step_one",
                "step_two",
                "step_three",
                "step_four",
                "step_five",
              ],
              "title": "Tomato soup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
            <Recipe
              avatarURL=""
              desc="Preparation time: 20min, Difficulty: 2/5"
              dislikes={5}
              handleEditRecipe={[Function]}
              handleRemoveRecipe={[Function]}
              id={2}
              ingredients={
                Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ]
              }
              likes={5}
              procedure={
                Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ]
              }
              title="Tomato soup"
            />
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Recipe
                avatarURL=""
                desc="Preparation time: 20min, Difficulty: 2/5"
                dislikes={5}
                handleEditRecipe={[Function]}
                handleRemoveRecipe={[Function]}
                id={2}
                ingredients={
                  Array [
                    "one",
                    "two",
                    "three",
                    "four",
                    "five",
                  ]
                }
                likes={5}
                procedure={
                  Array [
                    "step_one",
                    "step_two",
                    "step_three",
                    "step_four",
                    "step_five",
                  ]
                }
                title="Tomato soup"
              />
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <Recipe
                avatarURL=""
                desc="Preparation time: 20min, Difficulty: 2/5"
                dislikes={5}
                handleEditRecipe={[Function]}
                handleRemoveRecipe={[Function]}
                id={2}
                ingredients={
                  Array [
                    "one",
                    "two",
                    "three",
                    "four",
                    "five",
                  ]
                }
                likes={5}
                procedure={
                  Array [
                    "step_one",
                    "step_two",
                    "step_three",
                    "step_four",
                    "step_five",
                  ]
                }
                title="Tomato soup"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "avatarURL": "",
                "desc": "Preparation time: 20min, Difficulty: 2/5",
                "dislikes": 5,
                "handleEditRecipe": [Function],
                "handleRemoveRecipe": [Function],
                "id": 2,
                "ingredients": Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ],
                "likes": 5,
                "procedure": Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ],
                "title": "Tomato soup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<RecipeList /> Check if rendering recipes works correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeList
    filterText=""
    handleEditRecipe={[Function]}
    handleRemoveRecipe={[Function]}
    recipes={
      Array [
        Object {
          "avatarURL": "",
          "desc": "Preparation time: 30min, Difficulty: 3/5",
          "dislikes": 4,
          "id": 1,
          "ingredients": Array [
            "one",
            "two",
            "three",
            "four",
            "five",
          ],
          "likes": 4,
          "procedure": Array [
            "step_one",
            "step_two",
            "step_three",
            "step_four",
            "step_five",
          ],
          "title": "Spaghetti",
        },
        Object {
          "avatarURL": "",
          "desc": "Preparation time: 20min, Difficulty: 2/5",
          "dislikes": 5,
          "id": 2,
          "ingredients": Array [
            "one",
            "two",
            "three",
            "four",
            "five",
          ],
          "likes": 5,
          "procedure": Array [
            "step_one",
            "step_two",
            "step_three",
            "step_four",
            "step_five",
          ],
          "title": "Tomato soup",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
          <Recipe
            avatarURL=""
            desc="Preparation time: 30min, Difficulty: 3/5"
            dislikes={4}
            handleEditRecipe={[Function]}
            handleRemoveRecipe={[Function]}
            id={1}
            ingredients={
              Array [
                "one",
                "two",
                "three",
                "four",
                "five",
              ]
            }
            likes={4}
            procedure={
              Array [
                "step_one",
                "step_two",
                "step_three",
                "step_four",
                "step_five",
              ]
            }
            title="Spaghetti"
          />
        </li>
        <li>
          <Recipe
            avatarURL=""
            desc="Preparation time: 20min, Difficulty: 2/5"
            dislikes={5}
            handleEditRecipe={[Function]}
            handleRemoveRecipe={[Function]}
            id={2}
            ingredients={
              Array [
                "one",
                "two",
                "three",
                "four",
                "five",
              ]
            }
            likes={5}
            procedure={
              Array [
                "step_one",
                "step_two",
                "step_three",
                "step_four",
                "step_five",
              ]
            }
            title="Tomato soup"
          />
        </li>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <Recipe
              avatarURL=""
              desc="Preparation time: 30min, Difficulty: 3/5"
              dislikes={4}
              handleEditRecipe={[Function]}
              handleRemoveRecipe={[Function]}
              id={1}
              ingredients={
                Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ]
              }
              likes={4}
              procedure={
                Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ]
              }
              title="Spaghetti"
            />
          </li>,
          <li>
            <Recipe
              avatarURL=""
              desc="Preparation time: 20min, Difficulty: 2/5"
              dislikes={5}
              handleEditRecipe={[Function]}
              handleRemoveRecipe={[Function]}
              id={2}
              ingredients={
                Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ]
              }
              likes={5}
              procedure={
                Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ]
              }
              title="Tomato soup"
            />
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <Recipe
              avatarURL=""
              desc="Preparation time: 30min, Difficulty: 3/5"
              dislikes={4}
              handleEditRecipe={[Function]}
              handleRemoveRecipe={[Function]}
              id={1}
              ingredients={
                Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ]
              }
              likes={4}
              procedure={
                Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ]
              }
              title="Spaghetti"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "avatarURL": "",
              "desc": "Preparation time: 30min, Difficulty: 3/5",
              "dislikes": 4,
              "handleEditRecipe": [Function],
              "handleRemoveRecipe": [Function],
              "id": 1,
              "ingredients": Array [
                "one",
                "two",
                "three",
                "four",
                "five",
              ],
              "likes": 4,
              "procedure": Array [
                "step_one",
                "step_two",
                "step_three",
                "step_four",
                "step_five",
              ],
              "title": "Spaghetti",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Recipe
              avatarURL=""
              desc="Preparation time: 20min, Difficulty: 2/5"
              dislikes={5}
              handleEditRecipe={[Function]}
              handleRemoveRecipe={[Function]}
              id={2}
              ingredients={
                Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ]
              }
              likes={5}
              procedure={
                Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ]
              }
              title="Tomato soup"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "avatarURL": "",
              "desc": "Preparation time: 20min, Difficulty: 2/5",
              "dislikes": 5,
              "handleEditRecipe": [Function],
              "handleRemoveRecipe": [Function],
              "id": 2,
              "ingredients": Array [
                "one",
                "two",
                "three",
                "four",
                "five",
              ],
              "likes": 5,
              "procedure": Array [
                "step_one",
                "step_two",
                "step_three",
                "step_four",
                "step_five",
              ],
              "title": "Tomato soup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
            <Recipe
              avatarURL=""
              desc="Preparation time: 30min, Difficulty: 3/5"
              dislikes={4}
              handleEditRecipe={[Function]}
              handleRemoveRecipe={[Function]}
              id={1}
              ingredients={
                Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ]
              }
              likes={4}
              procedure={
                Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ]
              }
              title="Spaghetti"
            />
          </li>
          <li>
            <Recipe
              avatarURL=""
              desc="Preparation time: 20min, Difficulty: 2/5"
              dislikes={5}
              handleEditRecipe={[Function]}
              handleRemoveRecipe={[Function]}
              id={2}
              ingredients={
                Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ]
              }
              likes={5}
              procedure={
                Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ]
              }
              title="Tomato soup"
            />
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Recipe
                avatarURL=""
                desc="Preparation time: 30min, Difficulty: 3/5"
                dislikes={4}
                handleEditRecipe={[Function]}
                handleRemoveRecipe={[Function]}
                id={1}
                ingredients={
                  Array [
                    "one",
                    "two",
                    "three",
                    "four",
                    "five",
                  ]
                }
                likes={4}
                procedure={
                  Array [
                    "step_one",
                    "step_two",
                    "step_three",
                    "step_four",
                    "step_five",
                  ]
                }
                title="Spaghetti"
              />
            </li>,
            <li>
              <Recipe
                avatarURL=""
                desc="Preparation time: 20min, Difficulty: 2/5"
                dislikes={5}
                handleEditRecipe={[Function]}
                handleRemoveRecipe={[Function]}
                id={2}
                ingredients={
                  Array [
                    "one",
                    "two",
                    "three",
                    "four",
                    "five",
                  ]
                }
                likes={5}
                procedure={
                  Array [
                    "step_one",
                    "step_two",
                    "step_three",
                    "step_four",
                    "step_five",
                  ]
                }
                title="Tomato soup"
              />
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <Recipe
                avatarURL=""
                desc="Preparation time: 30min, Difficulty: 3/5"
                dislikes={4}
                handleEditRecipe={[Function]}
                handleRemoveRecipe={[Function]}
                id={1}
                ingredients={
                  Array [
                    "one",
                    "two",
                    "three",
                    "four",
                    "five",
                  ]
                }
                likes={4}
                procedure={
                  Array [
                    "step_one",
                    "step_two",
                    "step_three",
                    "step_four",
                    "step_five",
                  ]
                }
                title="Spaghetti"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "avatarURL": "",
                "desc": "Preparation time: 30min, Difficulty: 3/5",
                "dislikes": 4,
                "handleEditRecipe": [Function],
                "handleRemoveRecipe": [Function],
                "id": 1,
                "ingredients": Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ],
                "likes": 4,
                "procedure": Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ],
                "title": "Spaghetti",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <Recipe
                avatarURL=""
                desc="Preparation time: 20min, Difficulty: 2/5"
                dislikes={5}
                handleEditRecipe={[Function]}
                handleRemoveRecipe={[Function]}
                id={2}
                ingredients={
                  Array [
                    "one",
                    "two",
                    "three",
                    "four",
                    "five",
                  ]
                }
                likes={5}
                procedure={
                  Array [
                    "step_one",
                    "step_two",
                    "step_three",
                    "step_four",
                    "step_five",
                  ]
                }
                title="Tomato soup"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "avatarURL": "",
                "desc": "Preparation time: 20min, Difficulty: 2/5",
                "dislikes": 5,
                "handleEditRecipe": [Function],
                "handleRemoveRecipe": [Function],
                "id": 2,
                "ingredients": Array [
                  "one",
                  "two",
                  "three",
                  "four",
                  "five",
                ],
                "likes": 5,
                "procedure": Array [
                  "step_one",
                  "step_two",
                  "step_three",
                  "step_four",
                  "step_five",
                ],
                "title": "Tomato soup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
